test/unit/core/algorithm/configuration_test.cpp

```diff
@@ -222,58 +223,139 @@ TEST(configuration, remove_by_type_template)
     }
 }
 
-TEST(configuration, value_or_by_type)
+TEST(configuration, get_or_by_type)
 {
     seqan3::configuration cfg = bax{2.2} | bar{1};
 
     { // l-value
-        EXPECT_FLOAT_EQ(cfg.value_or<bax>(1.3), 2.2);
-        EXPECT_FLOAT_EQ(cfg.value_or<foo>(1.3), 1.3);
+        EXPECT_FLOAT_EQ(cfg.get_or(bax{1.3}).value, 2.2);
+        EXPECT_TRUE((std::same_as<decltype(cfg.get_or(bax{1.3})), bax &>));
+        EXPECT_EQ(cfg.get_or(foo{"test"}).value, "test");
     }
 
     { // const l-value
-        seqan3::configuration<bax, bar> const cfg_c{cfg};
-        EXPECT_FLOAT_EQ(cfg_c.value_or<bax>(1.3), 2.2);
-        EXPECT_FLOAT_EQ(cfg_c.value_or<foo>(1.3), 1.3);
+        EXPECT_FLOAT_EQ(std::as_const(cfg).get_or(bax{1.3}).value, 2.2);
+        EXPECT_TRUE((std::same_as<decltype(std::as_const(cfg).get_or(bax{1.3})), bax const &>));
+        EXPECT_EQ(std::as_const(cfg).get_or(foo{"test"}).value, "test");
     }
 
     { // r-value
-        seqan3::configuration<bax, bar> cfg_r{cfg};
-        EXPECT_FLOAT_EQ(std::move(cfg_r).value_or<bax>(1.3), 2.2);
-        EXPECT_FLOAT_EQ(std::move(cfg_r).value_or<foo>(1.3), 1.3);
+        EXPECT_FLOAT_EQ(std::move(seqan3::configuration<bax, bar>{cfg}).get_or(bax{1.3}).value, 2.2);
+        EXPECT_TRUE((std::same_as<decltype(seqan3::configuration<bax, bar>{cfg}.get_or(bax{1.3})), bax &&>));
+        EXPECT_EQ(std::move(seqan3::configuration<bax, bar>{cfg}).get_or(foo{"test"}).value, "test");
     }
 
     { // const r-value
         seqan3::configuration<bax, bar> const cfg_cr{cfg};
-        EXPECT_FLOAT_EQ(std::move(cfg_cr).value_or<bax>(1.3), 2.2);
-        EXPECT_FLOAT_EQ(std::move(cfg_cr).value_or<foo>(1.3), 1.3);
+        EXPECT_FLOAT_EQ(std::move(cfg_cr).get_or(bax{1.3}).value, 2.2);
+        // TODO: Enable when gcc7 support is dropped or seqan3::get is implemented as CPO.
+        // seqan3::configuration<bax, bar> const cfg_cr2{cfg};
+        // EXPECT_TRUE((std::same_as<decltype(std::move(cfg_cr2).get_or(bax{1.3})), bax const &&>));
+        seqan3::configuration<bax, bar> const cfg_cr3{cfg};
+        EXPECT_EQ(std::move(cfg_cr3).get_or(foo{"test"}).value, "test");
     }
 }
 
-TEST(configuration, value_or_by_type_template)
+TEST(configuration, get_or_by_type_constexpr)
 {
-    seqan3::configuration cfg = bar{1} | foobar<>{std::vector<int>{0, 1, 2, 3}};
+    constexpr seqan3::configuration cfg = bax{2.2} | bar{1};
 
     { // l-value
-        EXPECT_TRUE(ranges::equal(cfg.value_or<foobar>(3.3), std::vector{0, 1, 2, 3}));
-        EXPECT_FLOAT_EQ(cfg.value_or<foo>(1.3), 1.3);
+        constexpr auto element = cfg.get_or(bax{1.3});
+        EXPECT_FLOAT_EQ(element.value, 2.2);
```

Comment:
Why is the type not tested here as done in the tests above? Could we use a typed test here?

Reply:
Well, it better should be :)